src/main/java/net/fabricmc/loom/util/GradleSupport.java

```diff
@@ -34,24 +34,6 @@
 //This is used to bridge the gap over large gradle api changes.
 public class GradleSupport {
 	public static RegularFileProperty getfileProperty(Project project) {
-		try {
-			//First try the new method, if that fails fall back.
-			return getfilePropertyModern(project);
-		} catch (Exception e) {
-			//Nope
-		}
-
-		return getfilePropertyLegacy(project);
-	}
-
-	private static RegularFileProperty getfilePropertyModern(Project project) throws NoSuchMethodException, InvocationTargetException, IllegalAccessException {
-		ObjectFactory objectFactory = project.getObjects();
-		Method method = objectFactory.getClass().getDeclaredMethod("fileProperty");
-		method.setAccessible(true);
-		return (RegularFileProperty) method.invoke(objectFactory);
-	}
-
-	private static RegularFileProperty getfilePropertyLegacy(Project project) {
-		return project.getLayout().fileProperty();
+		return project.getObjects().fileProperty();
```

Comment:
This class could be removed completely if there's no backwards compatibility being gained from it anymore

Reply:
Yep, should remove this completely.

Reply:
Okey will do.